### PR TITLE
Fix attributes passed to form has_many not being set on new record form items

### PR DESCRIPTION
**Pull request by Fs00**
This PR fixes a behavior inconsistency that occurs when passing custom attributes to the form builder `has_many` method, like in the following example:
```ruby
f.has_many :posts, data: { controller: "custom" } do |p|
  ...
end
```
which builds the following HTML (I kept only the relevant parts):
```html
<li class="has_many_container posts" data-sortable-start="0">
  <fieldset data-controller="custom" class="inputs has_many_fields">...</fieldset>
   ...
  <fieldset data-controller="custom" class="inputs has_many_fields">...</fieldset>
   ...
</li>
```
However, when clicking the "New Record" link to add a new item to the association, the `<fieldset>` generated for the new item didn't have the user-provided attributes on it (it was just `<fieldset class="inputs has_many_fields">...</fieldset>`).

I've discovered that the HTML used for new items is stored in the `data-html` attribute of the link and generated by a different method that was ignoring all custom attributes except for `class`.
So I've changed the method to make sure that custom attributes are not ignored and added tests for the relevant scenarios.

We'd need this fix on a 3.x patch, should I open the backport PR once this is merged or will you take care of it?

### CODE REVIEW

**Comment by javierjulio** [comment]
@Fs00 this looks great, thank you. Yes, you should open a backport PR on the 3-0-stable branch please. My focus is on v4. Usually another maintainer will pick up doing a v3 release. We have at least another change considered to do on v3 regarding Sass deprecations. Either way I will approve and merge that PR if you create it. In the meantime, I'll merge this. Thank you.

**Review by javierjulio — APPROVED** [review]

**Review by javierjulio — APPROVED** [review]
Thanks!

**Comment by Fs00** [comment]
Thank you @javierjulio. I have opened #8551 to backport the change to AA 3.

**Pull request merged**
